cmd/skaffold/app/cmd/flags.go

```diff
@@ -66,7 +66,7 @@ var FlagRegistry = []Flag{
 		Shorthand:     "n",
 		Usage:         "Run deployments in the specified namespace",
 		Value:         &opts.Namespace,
-		DefValue:      "",
+		DefValue:      "default",
```

Comment:
let's separate this out to a different PR with tests and everything, I'm not sure about that this is what we want to do (we might but I need a bit more time to think it through).

Reply:
more thought around this: there is a difference between leaving it empty and letting the resources determine their namespace vs enforcing `default`.